File: 94. Binary Tree Inorder Traversal.java

import java.util.List;

class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        solve(res, root);
        return res;
    }
    void solve(List<Integer> res, TreeNode root){
        if(root != null){
            solve(res, root.left);
            res.add(root.val);
            solve(res, root.right);
        }
    }
}

File: 94. Binary Tree Inorder Traversal.go
func inorderTraversal(root *TreeNode) []int {
	res := int[]
	if root == nil{ return res}

	stack := *TreeNode[]{}
	for root != nil{
		stack = apend(stack, root)
		root = root.Left
	}
	for len(stack) > 0{
		node := stack[len(stack) - 1]
		res = apend(res, node.Val)
	}
	return res;
}